python/lsst/pipe/tasks/dcrAssembleCoadd.py

```diff
@@ -499,25 +485,39 @@ def calculateNImage(self, dcrModels, bbox, tempExpRefList, spanSetMaskList, stat
         -------
         dcrNImages : `list` of `lsst.afw.image.ImageU`
             List of exposure count images for each subfilter
+        dcrWeights : `list` of `lsst.afw.image.ImageF`
+            Per-pixel weights for each subfilter.
+            Equal to the number of unmasked images contributing to each pixel.
         """
         dcrNImages = [afwImage.ImageU(bbox) for subfilter in range(self.config.dcrNumSubfilters)]
+        dcrWeights = [afwImage.ImageF(bbox) for subfilter in range(self.config.dcrNumSubfilters)]
         tempExpName = self.getTempExpDatasetName(self.warpType)
         for tempExpRef, altMaskSpans in zip(tempExpRefList, spanSetMaskList):
             exposure = tempExpRef.get(tempExpName + "_sub", bbox=bbox)
+            # padValue = afwImage.pixel.SinglePixelF(0., exposure.mask.getPlaneBitMask("NO_DATA"), 0)
             visitInfo = exposure.getInfo().getVisitInfo()
             wcs = exposure.getInfo().getWcs()
             mask = exposure.mask
             if altMaskSpans is not None:
                 self.applyAltMaskPlanes(mask, altMaskSpans)
+            weightImage = np.zeros_like(exposure.image.array)
+            weightImage[mask.array & statsCtrl.getAndMask() == 0] = 1.
             dcrShift = calculateDcr(visitInfo, wcs, dcrModels.filter, self.config.dcrNumSubfilters)
-            for dcr, dcrNImage in zip(dcrShift, dcrNImages):
-                shiftedImage = applyDcr(exposure.maskedImage, dcr, self.warpCtrl, useInverse=True)
-                dcrNImage.array[shiftedImage.mask.array & statsCtrl.getAndMask() == 0] += 1
-        return dcrNImages
-
-    def dcrAssembleSubregion(self, dcrModels, bbox, dcrBBox, tempExpRefList, imageScalerList, weightList,
-                             spanSetMaskList, statsFlags, statsCtrl, convergenceMetric,
-                             baseMask, gain, modelWeights, refImage):
+            for dcr, dcrNImage, dcrWeight in zip(dcrShift, dcrNImages, dcrWeights):
+                shiftedWeights = applyDcr(weightImage, dcr, useInverse=True,
+                                          order=self.config.maskInterpOrder)
+                dcrNImage.array += np.rint(shiftedWeights).astype(dcrNImage.array.dtype)
```

Comment:
Sadly it casts to the wrong type of `int` and throws an error. I'm not very happy with how this is implemented, so I'd welcome any suggestions you might have.

Reply:
"wrong type of int"? What is the type of `dcrNImage`? And what is the error?

Reply:
The error if `.astype` is omitted is `TypeError: Cannot cast ufunc add output from dtype('float32') to dtype('uint16'), since `rint` returns the values rounded to the nearest integer as `float`s, but my comment about the "wrong type or int" came from setting `np.rint(...).astype(int)`, which raises `TypeError: Cannot cast ufunc add output from dtype('int64') to dtype('uint16') with casting rule 'same_kind'`.
So, I think I need to either leave this as it is, or add a completely different calculation of the nImage.

Reply:
Huh. Yeah, lets leave it as is and we can revisit when we look at performance.